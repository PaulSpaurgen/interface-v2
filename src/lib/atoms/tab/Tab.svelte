<script lang="ts">
	import { doNothing } from '$lib/utils/helpers/commonHelper';

	export let onclick = () => {
		doNothing();
	};
	export let styleClass = '';
	export let isActive = false;

	//reactive value based on isActive and styleClass
	$: buttonStyle = `${
		isActive ? 'bg-blue-100 text-primary font-medium' : 'bg-base-300 text-grey'
	} ${styleClass}`;
</script>

<button on:click={onclick} class="w-32 rounded-2xl py-1 text-sm hover:font-medium {buttonStyle}">
	<slot />
</button>
