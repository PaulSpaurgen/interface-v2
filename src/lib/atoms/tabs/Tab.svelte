<script lang="ts">
	export let id: string;
	export let buttonClass = '';
	export let liClass = 'mr-2';
</script>

<li class={liClass} role="presentation" data-testid="tab">
	<button on:click class={buttonClass} id="{id}-tabhead" type="button" role="tab">
		<slot />
	</button>
</li>
