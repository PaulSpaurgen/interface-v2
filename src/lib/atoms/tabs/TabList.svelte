<script>
	export let divClass = '';
</script>

<div class={divClass}>
	<ul class="-mb-px flex flex-wrap items-center justify-center gap-1" role="tablist">
		<slot />
	</ul>
</div>
