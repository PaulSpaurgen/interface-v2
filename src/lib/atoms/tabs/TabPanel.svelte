<script lang="ts">
	export let activeTabValue: string;
	export let id: string;
</script>

{#if activeTabValue === id}
	<div id="{id}-tabitem" role="tabpanel" aria-labelledby="{id}-tab" data-testid="tab-panel">
		<slot />
	</div>
{/if}
