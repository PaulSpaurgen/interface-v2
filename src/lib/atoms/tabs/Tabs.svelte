<script lang="ts">
	export let divClass = 'w-full';
</script>

<div class={divClass}>
	<slot />
</div>
