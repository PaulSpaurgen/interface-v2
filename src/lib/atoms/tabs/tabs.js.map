{"version":3,"file":"tabs.js","sourceRoot":"","sources":["tabs.ts"],"names":[],"mappings":"AAAA,OAAO,EAAE,OAAO,IAAI,IAAI,EAAE,MAAM,eAAe,CAAC;AAChD,OAAO,EAAE,OAAO,IAAI,OAAO,EAAE,MAAM,kBAAkB,CAAC;AACtD,OAAO,EAAE,OAAO,IAAI,QAAQ,EAAE,MAAM,mBAAmB,CAAC;AACxD,OAAO,EAAE,OAAO,IAAI,GAAG,EAAE,MAAM,cAAc,CAAC"}