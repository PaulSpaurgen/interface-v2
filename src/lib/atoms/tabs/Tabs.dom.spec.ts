import { cleanup, render } from '@testing-library/svelte';
import Tab from './Tab.svelte';
import { TabList, TabPanel, Tabs } from './tabs';

describe('Tabs', () => {
	afterEach(() => {
		cleanup();
		vi.resetAllMocks();
	});

	test('tab Render', () => {
		const { getByTestId } = render(Tab, { props: { id: 'tab_test' } });
		const tabElement = getByTestId('tab');
		expect(tabElement.childNodes).toHaveLength(1);
		expect((tabElement.firstChild as Element)?.id === 'tab_test-tabhead').toBe(true);
		expect(tabElement).toMatchSnapshot();
	});

	test('tab Render check using getByTestId & getByRole ', () => {
		const { getByTestId, getByRole } = render(Tab, { props: { id: 'tab_test' } });
		expect(getByTestId('tab')).toEqual(getByRole('presentation'));
	});

	test('tab Render with all atributes', () => {
		const { getByTestId } = render(Tab, {
			props: { buttonClass: 'primary-class', liClass: 'list', id: 'tab_test' }
		});
		const tabElement = getByTestId('tab');
		expect(tabElement.classList.contains('list')).toBeTruthy();
		expect((tabElement.firstChild as Element)?.classList.contains('primary-class')).toBe(true);
	});

	test('Default tabs Render', () => {
		const { getByTestId } = render(Tabs);
		const tabElement = getByTestId('tabs');
		expect(tabElement.classList.contains('w-full')).toBe(true);
	});

	test(' tabs render with divClass', () => {
		const { getByTestId } = render(Tabs, { props: { divClass: 'tabs-primary' } });
		const tabElement = getByTestId('tabs');
		expect(tabElement.classList.contains('tabs-primary')).toBe(true);
	});

	test('Default tablist render', () => {
		const { getByTestId, getByRole } = render(TabList);
		const tabListElement = getByTestId('tab-list');
		expect(tabListElement.contains(getByRole('tablist'))).toBe(true);
		expect(tabListElement).toMatchSnapshot();
	});

	test('tablist render divClass', () => {
		const { getByTestId } = render(TabList, { props: { divClass: 'tablist-divclass' } });
		const tabListElement = getByTestId('tab-list');
		expect(tabListElement.classList.contains('tablist-divclass')).toBe(true);
	});

	test('TabPanel not render', () => {
		const { queryAllByTestId } = render(TabPanel, { props: { activeTabValue: '10', id: '20' } });
		expect(queryAllByTestId('tab-panel')).toHaveLength(0);
	});

	test('TabPanel render', () => {
		const { getByTestId } = render(TabPanel, { props: { activeTabValue: '10', id: '10' } });
		expect(getByTestId('tab-panel').id === '10-tabitem').toBeTruthy();
		expect(getByTestId('tab-panel')).toMatchSnapshot();
	});

	test('TabPanel render using getByRole ', () => {
		const { getByRole } = render(TabPanel, { props: { activeTabValue: '50', id: '50' } });
		expect(getByRole('tabpanel').getAttribute('data-testid') === 'tab-panel').toBe(true);
	});
});
