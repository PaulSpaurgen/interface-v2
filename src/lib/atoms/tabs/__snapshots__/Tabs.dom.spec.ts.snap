// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tabs > Default tablist render 1`] = `
<div
  class=""
  data-testid="tab-list"
>
  <ul
    class="-mb-px flex flex-wrap items-center justify-center gap-1"
    role="tablist"
  />
</div>
`;

exports[`Tabs > TabPanel render 1`] = `
<div
  aria-labelledby="10-tab"
  data-testid="tab-panel"
  id="10-tabitem"
  role="tabpanel"
/>
`;

exports[`Tabs > tab Render 1`] = `
<li
  class="mr-2"
  data-testid="tab"
  role="presentation"
>
  <button
    class=""
    id="tab_test-tabhead"
    role="tab"
    type="button"
  />
</li>
`;
